### PR TITLE
refactor: re-order fields in JsInfo for readability

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

File: js/private/js_info.bzl [hunk 1]
```diff
@@ -3,53 +3,73 @@
...
         npm_package_store_infos: See JsInfo documentation
-        sources: See JsInfo documentation
-        transitive_types: See JsInfo documentation
-        transitive_sources: See JsInfo documentation
 
     Returns:
         A JsInfo provider
     """
+    if type(target) != "Label":
+        msg = "Expected target to be a Label but got {}".format(type(target))
```
Comment (jbedard) [inline js/private/js_info.bzl:42]: Why add the `msg` var everywhere?

**Comment by jbedard** [comment]
Why are these being added? What will they be used for?

**Comment by gregmagolan** [comment]
Read the summary

File: js/private/js_info.bzl (hunk 1 above)
Comment (gregmagolan) [reply js/private/js_info.bzl:42]: If you don't `msg` then Bazel shows the template string in the error message which is confusing. Using `msg` makes it show us as `fail(msg)` on the callstack when it is hit

File: js/private/js_helpers.bzl [hunk 2]
```diff
@@ -222,8 +222,8 @@ def gather_runfiles(
     transitive_files_depsets.append(gather_files_from_js_info(
         targets = data + deps,
         include_sources = include_sources,
-        include_transitive_sources = include_transitive_sources,
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply js/private/js_helpers.bzl:225]: Preference here. I found alphabetical harder to read since `sources` comes after all of the `transitive_*`

**Comment by gregmagolan** [comment]
> Why are these being added? What will they be used for?

Read the summary

**Comment by jbedard** [comment]
I know it's for #1646 and have looked at the code there as well, I was still hoping for more of an understanding of it though. There's no point merging this without understanding it first 🤷

File: js/private/js_helpers.bzl (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbedard) [reply js/private/js_helpers.bzl:225]: SGTM 👍

**Comment by gregmagolan** [comment]
> I know it's for #1646 and have looked at the code there as well, I was still hoping for more of an understanding of it though. There's no point merging this without understanding it first 🤷

The code in #1646 is fairly easy to follow to find the use case. TL;DR is that in #1646 we're linking to a directory in the output tree so there is no `File` to get the path to that from. Instead we use a combination of the JsInfo bin_dir and the target.package to determine the directory path in the output tree to symlink to.

**Comment by jbedard** [comment]
I would vote to only add these fields once we need them and they are actually used, so in the main PR. Want to change this PR just to do the re-ordering and `fail` cleanup?

**Comment by gregmagolan** [comment]
> I would vote to only add these fields once we need them and they are actually used, so in the main PR. Want to change this PR just to do the re-ordering and `fail` cleanup?

Sure. Updated.

**Review by jbedard — APPROVED** [review]

**Pull request merged**